inst/include/dust/interface.hpp

```diff
@@ -245,6 +285,15 @@ inline std::vector<size_t> r_index_to_index(cpp11::sexp r_index,
   return index;
 }
 
+inline std::vector<size_t> r_index_to_index_default(size_t n) {
+  std::vector<size_t> index;
+  index.reserve(n);
+  for (size_t i = 0; i < n; ++i) {
+    index.push_back(i);
+  }
```

Comment:
versus:
```
std::vector<size_t> index(n);
for (size_t i = 0; i < n; ++i) {
  index[i] = i;
}
```
which is preferred? Your way doesn't zero initialise/default construct the vector, but I often use mine as it's easier to use with openmp

Reply:
Also my preferred way of doing this:
```
#include <numeric>
std::vector<size_t> index(n);
std::iota(index.begin(), index.end(), 0);
```

Reply:
Lots of options for this trivial part of the PR it turns out

Reply:
I think they're all fine - with differences that probably become apparent if the code is in a performance-critical section. This will likely never get called the way that mcstate is shaping up! Do you feel strongly about the differences? In my mind the way it is clearest (least book-keeping, though the reserve micro-optiomisation could be removed), but I don't feel strongly at all

Reply:
Yeah keep it as is, just wanted to write them out to think if it made much of a difference as I'd done this a lot recently (the zero init is the main difference with a big object)